lib/ansible/modules/network/fortios/fortios_ipv4_policy.py

```diff
@@ -0,0 +1,324 @@
+        application_list          = dict(type='str'),
+    )
+
+    #merge global required_if & argument_spec from module_utils/fortios.py
+    argument_spec.update(fortios_argument_spec)
+    required_if = fortios_required_if
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=required_if,
+    )
+
+    result = dict(changed=False)
+
+    # fail if libs not present
+    msg = ""
+    if not HAS_PYFG:
+        module.fail_json(msg='Could not import the python library pyFG required by this module')
+
+    #define device
+    f = FortiOS( module.params['host'],
+        username=module.params['username'],
+        password=module.params['password'],
+        timeout=module.params['username'],
+        vdom=module.params['vdom'])
+
+    path = 'firewall policy'
+
+    #connect
+    try:
+        f.open()
+    except:
+        module.fail_json(msg='Error connecting device')
+
+    #get  config
+    try:
+        f.load_config(path=path)
+        result['firewall_address_config'] = f.running_config.to_text()
+    except:
+        module.fail_json(msg='Error reading running config')
+
+    #Absent State
+    if module.params['state'] == 'absent':
+        f.candidate_config[path].del_block(module.params['id'])
+        change_string = f.compare_config()
+        if change_string != "":
+            result['change_string'] = change_string
+            result['changed'] = True
+
+    #Present state
+    if module.params['state'] == 'present':
+        new_policy = FortiConfig(module.params['id'], 'edit')
+
+        #src / dest / service / interfaces
+        new_policy.set_param('srcintf', '"{0}"'.format(module.params['src_intf']))
+        new_policy.set_param('dstintf', '"{0}"'.format(module.params['dst_intf']))
+
+
+        new_policy.set_param('srcaddr', " ".join('"' + item + '"' for item in module.params['src_addr']))
+        new_policy.set_param('dstaddr', " ".join('"' + item + '"' for item in module.params['dst_addr']))
+        new_policy.set_param('service', " ".join('"' + item + '"' for item in module.params['service']))
+
+        # negate src / dest / service
+        if module.params['src_addr_negate']:
+            new_policy.set_param('srcaddr-negate', 'enable')
+        if module.params['dst_addr_negate']:
+            new_policy.set_param('dstaddr-negate', 'enable')
+        if module.params['service_negate']:
+            new_policy.set_param('service-negate', 'enable')
+
+        # action
+        new_policy.set_param('action', '{0}'.format(module.params['action']))
+
+        # Schedule
+        new_policy.set_param('schedule', '"{0}"'.format(module.params['schedule']))
+
+        #NAT
+        if module.params['nat']:
+            new_policy.set_param('nat', 'enable')
+            if module.params['fixedport']:
+                new_policy.set_param('fixedport', 'enable')
+            if module.params['poolname'] is not None:
```

Comment:
No.
But I need to test that nat is true when fixedport (or poolname) is set.
Will see how to do that